lib/topic_query.rb

```diff
@@ -269,6 +269,14 @@ def list_private_messages_group_archive(user)
     create_list(:private_messages, {}, list)
   end
 
+  def list_private_messages_tag(user)
+    list = private_messages_for(user, :all)
+    tag_id = Tag.where('name ilike ?', @options[:tags][0]).pluck(:id).first
```

Comment:
This is actually inefficient because it fetches all the `id`s into an array and discard the rest. It'll probably be better to just fetch one record from the DB and select the column you want. Maybe use `.select(:id)` if we're concern about forming an entire ActiveRecord object.

Reply:
I'm thinking if we are expecting a partial tag name considering that we are using `ilike` here.